Add missing sync in LINCS and SETTLE CUDA kernels
https://gitlab.com/gromacs/gromacs/-/merge_requests/2499
Patch added to EasyBuild by Simon Branford (University of Birmingham)
diff --git a/src/gromacs/mdlib/lincs_cuda.cu b/src/gromacs/mdlib/lincs_cuda.cu
index 27ad86f17f..4964a3d040 100644
--- a/src/gromacs/mdlib/lincs_cuda.cu
+++ b/src/gromacs/mdlib/lincs_cuda.cu
@@ -382,11 +382,15 @@ __launch_bounds__(c_maxThreadsPerBlock) __global__
                             sm_threadVirial[d * blockDim.x + (threadIdx.x + dividedAt)];
                 }
             }
-            // Syncronize if not within one warp
+            // Synchronize if not within one warp
             if (dividedAt > warpSize / 2)
             {
                 __syncthreads();
             }
+            else
+            {
+                __syncwarp();
+            }
         }
         // First 6 threads in the block add the results of 6 tensor components to the global memory address.
         if (threadIdx.x < 6)
diff --git a/src/gromacs/mdlib/settle_cuda.cu b/src/gromacs/mdlib/settle_cuda.cu
index c02f0d5884..c6ca1e2452 100644
--- a/src/gromacs/mdlib/settle_cuda.cu
+++ b/src/gromacs/mdlib/settle_cuda.cu
@@ -352,6 +352,10 @@ __launch_bounds__(c_maxThreadsPerBlock) __global__
             {
                 __syncthreads();
             }
+            else
+            {
+                __syncwarp();
+            }
         }
         // First 6 threads in the block add the 6 components of virial to the global memory address
         if (tib < 6)
